helix-core/src/main/java/org/apache/helix/util/InstanceValidationUtil.java

```diff
@@ -266,6 +269,9 @@ public static boolean isInstanceStable(HelixDataAccessor dataAccessor, String in
       IdealState idealState = dataAccessor.getProperty(keyBuilder.idealStates(idealStateName));
       if (idealState == null || !idealState.isEnabled() || !idealState.isValid()
           || TaskConstants.STATE_MODEL_NAME.equals(idealState.getStateModelDefRef())) {
+        _logger.warn(
+            "idealState is either null, not enabled, not valid, or has Task as stateModelDefRef. IdealState: {}",
+            instanceName);
```

Comment:
This won't make things fail, will it?

Reply:
While this block doesn't fail the code, it's triggered in situations where idealState is null or is invalid. Just like the logs I did in `InstanceServiceImpl.java`, I think it's a good idea to log these unusual situations for debug purposes.

Reply:
This block mostly exists to filter out non-existing IdealStates and Task Framework idealStates. The reason I pointed this out is because I don't think it will add much in debugging and it will just end up making the log file large. Let's not log this.

Reply:
I mostly want to ensure the case of `idealState == null` is caught. That situation shouldn't happen. 
Regardless, I don't have a strong opinion about this.

Reply:
You're not catching anything here, right? Also if IdealState is null, you skip over that resource, right? Would that ever cause a problem?